src/languages/c.coffee

```diff
@@ -3,42 +3,256 @@
+  'SharedIncludeLiteral': 'socket',
+  'Constant': 'socket'
 }
 
+COLOR_RULES = [
+  ['jumpStatement', 'return']
+  ['declaration', 'control'],
+  ['specialMethodCall', 'command'],
+  ['additiveExpression', 'value'],
+  ['multiplicativeExpression', 'value'],
+  ['postfixExpression', 'command'],
+  ['iterationStatement', 'control'],
+  ['selectionStatement', 'control'],
+  ['assignmentExpression', 'command'],
+  ['relationalExpression', 'value'],
+  ['initDeclarator', 'command'],
+  ['blockItemList', 'control'],
+  ['compoundStatement', 'control'],
+  ['externalDeclaration', 'control'],
+  ['structDeclaration', 'command'],
+  ['declarationSpecifier', 'control'],
+  ['statement', 'command'],
+  ['selectionStatement', 'control'],
+  ['iterationStatement', 'control'],
+  ['functionDefinition', 'control'],
+  ['expressionStatement', 'command'],
+  ['expression', 'value'],
+  ['parameterDeclaration', 'command'],
+  ['unaryExpression', 'value'],
+  ['typeName', 'value'],
+  ['initializer', 'value'],
+  ['castExpression', 'value'],
+]
+
+SHAPE_RULES = [
+  ['blockItem', 'block-only'],
+  ['expression', 'value-only'],
+  ['postfixExpression', 'block-only'],
+  ['equalityExpression', 'value-only'],
+  ['logicalAndExpression', 'value-only'],
+  ['logicalOrExpression', 'value-only'],
+  ['iterationStatement', 'block-only'],
+  ['selectionStatement', 'block-only'],
+  ['assignmentExpression', 'block-only'],
+  ['relationalExpression', 'value-only'],
+  ['initDeclarator', 'block-only'],
+  ['externalDeclaration', 'block-only'],
+  ['structDeclaration', 'block-only'],
+  ['declarationSpecifier', 'block-only'],
+  ['statement', 'block-only'],
+  ['selectionStatement', 'block-only'],
+  ['iterationStatement', 'block-only'],
+  ['functionDefinition', 'block-only'],
+  ['expressionStatement', 'value-only'],
+  ['expression', 'value-only'],
+  ['additiveExpression', 'value-only'],
+  ['multiplicativeExpression', 'value-only'],
+  ['declaration', 'block-only'],
+  ['parameterDeclaration', 'block-only'],
+  ['unaryExpression', 'value-only'],
+  ['typeName', 'value-only'],
+  ['initializer', 'value-only'],
+  ['castExpression', 'value-only']
+]
+
 config = {
-  INDENTS, SKIPS, PARENS, SOCKET_TOKENS, COLORS_FORWARD, COLORS_BACKWARD
+  RULES, COLOR_RULES, SHAPE_RULES
 }
 
+ADD_PARENS = (leading, trailing, node, context) ->
+  leading '(' + leading()
+  trailing trailing() + ')'
+
+config.PAREN_RULES = {
+  'primaryExpression': {
+    'expression': ADD_PARENS
+    'additiveExpression': ADD_PARENS
+    'multiplicativeExpression': ADD_PARENS
+    'assignmentExpression': ADD_PARENS
+    'postfixExpression': ADD_PARENS
+  }
+}
+
+config.SHOULD_SOCKET = (opts, node) ->
+  unless opts.knownFunctions? and ((node.parent? and node.parent.parent? and node.parent.parent.parent?) or
+      node.parent?.type is 'specialMethodCall')
+    return true
+
+  # If it is a function call, and we are the first child
+  if (node.parent.type is 'primaryExpression' and
```

Comment:
This is a very long set of conditions.  It deserves a comment showing an example expression or tree to justify what is being tested.